### ISSUE TITLE
New Weekly Note created on 2022-12-27 opens weekly note from 2022-01-01

### ISSUE THREAD

**Issue by christophberger** [liamcain/obsidian-periodic-notes#163]
Weekly notes worked fine throughout the year, until today when the New Weekly Note command opens the (existing) note from one year ago instead.

Configuration: `YYYY-MM-DD [Woche ]WW`

Expected: a new weekly note opens, with the file name "2022-12-27 Woche 52"

Observed: the existing weekly note "2022-01-01 Woche 52" opens

I skimmed the existing issues for a similar problem but other issues with weekly notes do not seem to be specific to the last week of the year.

The preview shows the correct expected file name:
<img width="786" alt="Screen Shot 2022-12-27 at 18 31 34" src="https://user-images.githubusercontent.com/3835890/209702671-2e449274-8dbc-4457-9b86-34f75be3935b.png">

The actual file name is this:
<img width="357" alt="Screen Shot 2022-12-27 at 18 31 52" src="https://user-images.githubusercontent.com/3835890/209702678-6da1d582-3339-4af2-a7e8-26745c5c5f38.png">

**Comment by jwhitley** [comment]
I'm seeing a probable variant of this as well. As of this week, instead of `Open weekly note` opening a note with filename `2023-Jan-15` (via my format of `gggg-MMM-DD`) the command opens the existing note from about year ago with filename `2022-Jan-24`.

**Comment by Wenzel** [comment]
Bumping this issue,
after a year of using Periodic Notes and Weekly Notes, I'm facing a case where the next Weekly note for W43 seems already created and circled back to 2022's W43.

![image](https://github.com/liamcain/obsidian-periodic-notes/assets/964610/e39dd65c-6d97-45b9-9d7d-b6147a2071a0)
![image](https://github.com/liamcain/obsidian-periodic-notes/assets/964610/9b8cbbd9-1bab-4a21-b063-d462158c7d4d)

**Comment by liamcain** [comment]
Weekly notes should always be using gggg (locale-aware) or GGGG (ISO) for the "week year" instead of "YYYY" which is the "year of the current day."


So your weekly note format should be:
GGGG-MM-DD [Woche ]WW

**Comment by Wenzel** [comment]
Hi @liamcain 

thanks for your reply today !
I just tried the fix you proposed:
![image](https://github.com/liamcain/obsidian-periodic-notes/assets/964610/5c95d15a-ba49-4caf-a1e8-48adec1c2f7e)

And I restarted Obsidian, but when I run the "Open Weekly Note" command:
![image](https://github.com/liamcain/obsidian-periodic-notes/assets/964610/5a861024-4401-4282-a4a6-3cb329752c8c)

I still get the one from 2022:
![image](https://github.com/liamcain/obsidian-periodic-notes/assets/964610/83e25e7f-e660-4aa8-b724-809c275eb4f4)

I have a feeling the problem is elsewhere.
Do you need more debug information ?

**Comment by liamcain** [comment]
@Wenzel you are mixing ISO formatting tokens with non-ISO formatting tokens. You can't use GGGG with ww. You either need to have `GGGG` with `WW` or `gggg` with `ww`.

**Comment by Wenzel** [comment]
Hi @liamcain,

Thanks for your feedback here, i really appreciate.
I converted my Weekly note file format to:
![image](https://github.com/liamcain/obsidian-periodic-notes/assets/964610/755334ae-1637-4c1e-ad56-2c093470b5c2)
Prefixing them by the current year, so they remain unique.

I have one last observation:
When I tried to convert my Daily notes to the ISO `GGGG` format, i get this result:
![image](https://github.com/liamcain/obsidian-periodic-notes/assets/964610/05a75818-5857-48bd-8cd0-6380e8816e02)

But my filename seems to be unique ? (`2023-11-07`)
So I'm confused about this message ?

When I keep the `YYYY` notation, Periodic Notes doesn't complain:
![image](https://github.com/liamcain/obsidian-periodic-notes/assets/964610/472d3187-1930-438a-8afd-be5a42ebaf63)

If i need to open a new issue about this, just tell me.
